### ISSUE TITLE
Unnecessarily hitting the 6 parallel connections limit of the browser (or some type of maximum connections limit?)

### ISSUE THREAD

**Issue by eecurfew** [caddyserver/caddy#6614]
## The issue:

I use Caddy as a reverse proxy, forwarding requests to another webserver. The app/webserver in question serves video files such as `.mp4`, `.webm` files, sort of like a personal instance of YouTube.

When I navigate relatively quickly between videos (in a single browser tab) I quite easily appear to hit the "maximum of 6 parallel connections" of the browser (Chromium). After which the video won't directly start playing, until it eventually starts after ~20 seconds. When this happens, it'll also stall when attempting to download any images that haven't been cached client-side already. 

All I do is:

1. Go to the homepage of my app;
2. Click on a video;
3. Press the back button;
4. Click on another video;
5. Repeat ~6 times, until a video won't start playing.
    - _The whole sequence takes maybe 6-10 seconds._

So, to me it seems like caddy keeps connections for video streams open for (too) long. Since I'm only playing a single video at a time (just switching quite rapidly).

## Attempt to fix:

I initially tried to fix the issue by adding `flush_interval -1` to the Caddyfile. I'm not entirely sure whether it improved things, but it definitely didn't fix the issue.

```Caddyfile
api.videos.localhost {
	reverse_proxy video-api_nginx:80 {
		flush_interval -1
	}
}
```

## Current workaround:

As a workaround I currently add 9 aliases, named `cdn-01.` - `cdn-09.`, and I assign each video a random subdomain. This _almost_ eliminates the issue, although it still randomly happen every now and than (lasts shorter though). So, my current Caddyfile is:

```Caddyfile
# Vite preview server
videos.localhost {
	redir https://www.videos.localhost
}
www.videos.localhost {
	reverse_proxy videos_preview:4173
}

# Vite dev server
www.dev.videos.localhost {
	redir https://dev.videos.localhost
}
dev.videos.localhost {
	reverse_proxy videos_node:5173
}

# API
www.api.videos.localhost {
	redir https://api.videos.localhost
}
api.videos.localhost,
cdn-01.api.videos.localhost,
cdn-02.api.videos.localhost,
cdn-03.api.videos.localhost,
cdn-04.api.videos.localhost,
cdn-05.api.videos.localhost,
cdn-06.api.videos.localhost,
cdn-07.api.videos.localhost,
cdn-08.api.videos.localhost,
cdn-09.api.videos.localhost {
	reverse_proxy video-api_nginx:80 {
		flush_interval -1
	}
}
```

## My Setup:

> I should mention that my setup is kind of "sketchy", so it could be the case that the bug is caused due to the relatively weird setup I'm running.
> 
> Although, that being said, I only recently started using Caddy (due to defaulting to `https` on localhost). Before this I used `nginx-proxy-manager`, which didn't ever run into this issue, even though my setup may be a bit weird.

I'm running Caddy in Docker (Compose) on Ubuntu 24.04 on WSL2 on Windows 11. Caddy serves as the main entry point for all my local apps as a reverse proxy. All apps are in the same docker network `caddy_default`. Caddy forwards all requests to another docker container that's still running `nginx`. The `nginx` docker container serves the files, which are stored in a bind mount that references the Windows' filesystem, like so:

```yml
volumes:
  - ./src:/var/www/html
  - /mnt/d/videos:/var/www/html/public/storage # <= /mnt/d is the `D:\` disk, which WSL2 mounts by default.
```

**Comment by francislavoie** [comment]
That sounds like a browser bug, not a problem with Caddy. If you navigate away from a page, the browser should be culling the connections.

**Comment by bt90** [comment]
Are you using HTTPS? Otherwise your browser won't use HTTP/2 or HTTP/3.

**Comment by mholt** [comment]
Yeah, if that is what is happening, it must be on HTTP/1.1 I imagine.

**Comment by eecurfew** [comment]
Appears to be a Chromium browser bug:

- I restored my old `nginx-proxy-manager` container, and I found out it also has the same issue (I just never noticed before).
- Firefox doesn't have this issue with either setup.

So I guess I just noticed it for the first time after I had only just moved to Caddy.

For what it's worth; I was using HTTPS + HTTP/2 with Caddy. While `nginx-proxy-manager` ran with HTTP + HTTP/1.1.

Sorry!
